### PR TITLE
Update to bevy 0.4

### PR DESCRIPTION
**Pull request by davidB**
Hi,

A small update to work with bevy 0.4, includes:

- upgrade of the dependencies into Cargo.toml, reduce the number of bevy features requires by the lib
- search and replace (into code, examples and doc) to be compatible with bevy 0.4
- change the README to prepare next release

### CODE REVIEW

**Comment by ootoovak** [comment]
I just confirmed this to work on my project ported from Bevy 0.3.0 to Bevy 0.4.0. Thanks @davidB!

File: Cargo.toml [hunk 1]
```diff
@@ -11,5 +11,8 @@ readme = "README.md"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-bevy = "0.3.0"
-lyon = "0.16.0"
+bevy = {version = "0.4.0", default-features = false, features=["render"]}
+lyon = "0.16.2"
```
Comment (ootoovak) [inline Cargo.toml:15]: ❓ Does this need version number need to be constrained so tightly to a patch version? Could it be loosened to be constrained to just the minor version number instead?

Comment (ootoovak) [reply Cargo.toml:15]: Same question for the `bevy` version number above.

**Comment by ootoovak** [comment]
I think a few crates use `.gitignore` for the `Cargo.lock` file as well to help with builds of people using the library. [One person here has also mentioned this.](https://github.com/Nilirad/bevy_prototype_lyon/pull/16#issuecomment-734362005) @Nilirad I wonder if we can also do this here?

**Comment by davidB** [comment]
I agree with both request,

- gitignore of `Cargo.lock`
- ignore patch number in dependencies

 because I'm lazy I didn't create a new PR, @Nilirad I can redo the PR if you like.

**Comment by Nilirad** [comment]
@davidB that's not a big deal, I'll do it alone in a couple of minutes.

**Pull request merged**